### PR TITLE
Promote Web Actions from Experimental

### PR DESCRIPTION
**Pull request by dubeejw**
- Promote `api/v1/experimental/web` to `api/v1/web`
- Allow CORS header to be set from Web Actions in `v1/web`
- Make extensions optional defaulting to `.http`
- Added .svg extension
- Inline body when it is not JSON
- Includes `raw-http` annotation support allowing Web Actions to receive query parameters and raw body content in `api/v1/web`
- Prevent invocation of an action multiple times from one request
- Display proper error when accept header type does not match actual return type

Closes: https://github.com/openwhisk/openwhisk/issues/1947
Closes: https://github.com/openwhisk/openwhisk/issues/1886
Related: https://github.com/openwhisk/openwhisk/issues/1965

### CODE REVIEW

**Comment by dubeejw** [comment]
PG2 1218, PG4 59.

**Comment by dubeejw** [comment]
PG4 50 approved.

**Comment by dubeejw** [comment]
PG1 1262, PG4 60.

**Comment by rabbah** [comment]
@markusthoemmes @cbickel do either of you have bandwidth to review these changes?

**Comment by starpit** [comment]
will /api/v1/experimental routes continue to work? we have apps that depend on this, and will need some grace period to port over to the non-experimental route

**Comment by dubeejw** [comment]
@starpit, the experimental route will remain intact for now.

**Comment by dubeejw** [comment]
PG4 65, approved.

File: core/controller/src/main/scala/whisk/core/controller/Meta.scala [hunk 1]
```diff
@@ -98,14 +190,21 @@ protected[core] object WhiskMetaApi extends Directives {
         extension: String,
         defaultProjection: Option[List[String]],
         projectionAllowed: Boolean,
-        transcoder: (JsValue, TransactionId) => RequestContext => Unit)
+        transcoder: (JsValue, TransactionId, WebApiDirectives) => RequestContext => Unit) {
+        val extensionLength = extension.length
+    }
 
-    private def resultAsHtml(result: JsValue, transid: TransactionId): RequestContext => Unit = result match {
+    private def resultAsHtml(result: JsValue, transid: TransactionId, rp: WebApiDirectives): RequestContext => Unit = result match {
```
Comment (cbickel) [inline core/controller/src/main/scala/whisk/core/controller/Meta.scala]: Why is `rp` needed here?

Comment (rabbah) [inline core/controller/src/main/scala/whisk/core/controller/Meta.scala]: fair question - and i tried to find a convenient workaround. we need to access the status code directives inside the response as http handler to determine the value of the "status code" field. the signature for all of these methods is the same, so for that reason, the directive is provided to all the methods. alternatively, one could create another closure with the status code field captured which seemed more complex and unnecessary.

Comment (cbickel) [inline core/controller/src/main/scala/whisk/core/controller/Meta.scala]: Okay, that's a fair point.

**Pull request merged**